js/controllers/grid/articleGalleys/ArticleGalleyGridHandler.js

```diff
@@ -58,11 +58,18 @@
 		// FIXME: Inter-widget messaging is needed here.
 		var selector = 'a[id^="component-grid-articlegalleys-articlegalleygrid-row-' +
 				rowId + '-addFile-button-"]';
-		$.when($(selector)).then(function() {
-			$(function() {
+
+		function defferedClick(selector) {
+			if ($(selector).length) {
 				$(selector).click();
-			});
-		});
+			} else {
+				setTimeout(function() {
```

Comment:
Rather than hard-coding an arbitrary wait time, why not use something like `$.when(...).then`? See e.g. https://github.com/asmecher/ojs/commit/e8ab81b6c372e5be53f4a70c0428194cd60a85e8

Reply:
This is reverting that change due to breaking changes introduced by jQuery 3.0: https://jquery.com/upgrade-guide/3.0/#breaking-change-and-feature-jquery-when-arguments. I don't think we can just throw a selector in there and wait for it to exist, like it was before. The `when` method expects a Promise (ES6-only) or jQuery's Deferred object. 

I can dig into jQuery's Deferred objects, but a quick scan yesterday suggested it would take more than a few minutes to get a handle on, so I opted for this route. I can dig into it more if you want.

Reply:
:disappointed: oops! I thought I had 3.0 after rebasing but it looks like it's still on 1.11. My bad! This whole PR is probably a mistake.